.gitignore

```diff
@@ -1,5 +1,20 @@
+.idea
+vendor/
+
+# composer
 composer.lock
 composer.phar
-vendor/
+
+# osx
 .DS_Store
-.idea
+.DS_Store?
+._*
+.Spotlight-V100
+.Trashes
+
+# npm
+.npm/node_modules
+node_modules
+
+# generated files
+templates/theme/css
```

Comment:
don't tell me what i wouldn't do! :smile: all kidding aside though, why wouldn't I? I usually test if scripts are working after any introduced changes, and cleaning up these dirs and files is a bit of a pain in the backside. any reason not to include those in .gitignore?